api/v1alpha1/bgppeer_webhook.go

```diff
@@ -102,17 +105,38 @@ func (bgpPeer *BGPPeer) validateBGPPeersRouterID(existingBGPPeersList *BGPPeerLi
 	return nil
 }
 
+func (bgpPeer *BGPPeer) validateBGPPeersMyASN(existingBGPPeersList *BGPPeerList) *field.Error {
+	myASN := bgpPeer.Spec.MyASN
+
+	for _, BGPPeer := range existingBGPPeersList.Items {
+		if myASN != BGPPeer.Spec.MyASN {
+			return field.Invalid(field.NewPath("spec").Child("MyASN"), myASN,
+				fmt.Sprintf("Multiple BGP instances not supported myASN %d existing myASN %d",
+					myASN, BGPPeer.Spec.MyASN))
+		}
+	}
+
+	return nil
+}
+
 func (bgpPeer *BGPPeer) validateBGPPeerConfig(existingBGPPeersList *BGPPeerList) *field.Error {
 	remoteASN := bgpPeer.Spec.ASN
+	myASN := bgpPeer.Spec.MyASN
 	address := bgpPeer.Spec.Address
+	srcAddr := bgpPeer.Spec.SrcAddress
 
-	if net.ParseIP(address) == nil {
+	if len(address) != 0 && net.ParseIP(address) == nil {
```

Comment:
Why would we need this? I think `net.ParseIP` of an empty string would still return nil

Reply:
I need to do the check/validtae only if the field is populated in the CR, if the field is empty I need to skip the test and not fail it

Reply:
But I think the peer address mandatory, otherwise we would not be able to establish a session.

Reply:
yes , i will limit the len check for srcaddress which is optional